spec/lita/adapters/irc/cinch_plugin_spec.rb

```diff
@@ -98,4 +101,50 @@
       subject.on_room_join(m)
     end
   end
+
+  describe "#on_room_part" do
+    before do
+      allow(Lita::Room).to receive(:create_or_update).and_return(room)
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
+    end
+
+    it "triggers a part event when someone parts a channel" do
+      expect(robot).to receive(:trigger).with(
+        :user_parted_room,
+        user: user,
+        room: room,
+      )
+      subject.on_room_part(m)
+    end
+  end
+
+  describe "#on_quit" do
+    before do
+      allow(Lita::User).to receive(:find_by_name).and_return(user)
+    end
+
+    it "triggers a disconnect event when someone quits from the IRC server" do
+      expect(robot).to receive(:trigger).with(
+        :user_disconnected,
+        user: user,
+      )
+      subject.on_quit(m)
+    end
+  end
+
+  describe "#on_nick_change" do
+    before do
+      allow(Lita::User).to receive(:find_by_name).and_return(from_user, user)
+    end
+
+    it "triggers a nick changed event when someone changes nick" do
+      expect(robot).to receive(:trigger).with(
+        :user_nick_changed,
+        from_user: from_user,
+        user: user,
+      )
+      expect(from_user.name).to eq(prefix.gsub(/!.*/, ''))
```

Comment:
You don't have to duplicate the implementation, you just want to ensure that a specific input produces a specific output. Given that, I think it's okay to use hardcoded strings instead of trying to dynamically create the expected output within the test.